jerry-port/default/defaultx-extfunc.c

```diff
@@ -0,0 +1,29 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+
+#include "jerryscript-ext/extfunc.h"
+
+/**
+ * Print a single character to stdout.
+ *
+ * @param c the character to print.
+ */
+void
+jerryx_port_extfunc_print_char (char c) /**< the character to print */
+{
+  printf ("%c", c);
```

Comment:
We are just removing that in #1749 .

Moreover, the needs of the external print handler are a bit different from the print implementation that we had as a builtin. (The builtin expected a printf-like vararg interface from jerry_port_console, but the current external print handler needs to write a single char only -- which IMHO is good, as vararg turned out to be a pain sometimes and for some targets.)

Moreover 2, jerry_port_console suggests that all ports implement it, but that's not the case. If an app does not use jerry-ext, it will not need to implement jerry_port_console.